### PR TITLE
Fix localStorage detection in Firefox

### LINKED ISSUE #2446: `getLocalStorageItems` can throw exception in Firefox
We've been receiving error reports for our browser extension ([Hypothesis](https://github.com/hypothesis/client)) caused by inline scripts that Privacy Badger injects into iframes to do with supercookie reporting.

The `getLocalStorageItems` function in Privacy Badger tests whether `localStorage` is accessible by just trying to [access the `window.localStorage` property](https://github.com/EFForg/privacybadger/blob/22f4b26bf308d00651e6b2ba797ce326a1985200/src/js/contentscripts/supercookie.js#L33). It seems that in Firefox there are scenarios where this property is readable but returns `null`. This causes errors later on when `getLocalStorageItems` tries to access this value as an object. One way to force this is to happen is to set the `dom.storage.enabled` flag in `about:config` to false.

Sentry report from our app: https://sentry.io/share/issue/4ee0ff95023445d4b427f598f8d2cd9d/

We're just going to ignore/suppress the error on our end to avoid distracting from errors that actually cause problems for users, but I thought this might be useful for the Privacy Badger team to be aware of.

### PR DESCRIPTION
**Pull request by hichamelkaddioui**
This should fix #2446

### CODE REVIEW

**Comment by ghostwords** [comment]
Hello and thank you for the PR! Looks like tests that have to do with detecting localStorage tracking are failing, which suggests localStorage detection may be broken.

**Comment by hichamelkaddioui** [comment]
Hi, after noticing the failing build, I setup functional tests on local, and I'm working on a fix!

**Comment by ghostwords** [comment]
By the way, you might want to fix your git/GitHub config to get your commits properly attributed to your GitHub account: https://help.github.com/en/articles/setting-your-commit-email-address

File: src/js/contentscripts/supercookie.js [hunk 1]
```diff
@@ -28,12 +28,18 @@ function getScPageScript() {
 
   // return a string
   return "(" + function () {
-
```
[the remark this answers is not in the document]
Comment (hichamelkaddioui) [reply src/js/contentscripts/supercookie.js]: Will do!

**Comment by hichamelkaddioui** [comment]
> By the way, you might want to fix your git/GitHub config to get your commits properly attributed to your GitHub account: https://help.github.com/en/articles/setting-your-commit-email-address

Whoopsies, thanks for the heads up!

> Shall we also apply the fix to [`clobberlocalstorage.js`](https://github.com/EFForg/privacybadger/blob/22f4b26bf308d00651e6b2ba797ce326a1985200/src/js/contentscripts/clobberlocalstorage.js#L41-L47)? (The logic there is applied to third-party frames belonging to yellowlisted domains, whereas the logic in `supercookie.js` is applied inside every third-party frame.)

I've added the same portion of code to `src/js/contentscripts/clobberlocalstorage.js` as well, thanks for the explanation.

At some point I found it tempting to factor the `localStorage` detection (though two occurrences is very acceptable), but since it's inside a stringified anonymous function, I'm not even sure that's doable in a simple way.

**Comment by hichamelkaddioui** [comment]
Looks like `dpkg` failed in the CI task for Chrome Beta :cry: 
```
dpkg-deb: error: archive '/tmp/google-chrome-beta_current_amd64.deb' has premature member 'control.tar.xz' before 'control.tar.gz', giving up
```

**Comment by ghostwords** [comment]
>Looks like dpkg failed in the CI task for Chrome Beta

That's fine, nothing to do with this PR, hopefully a one-off.

**Pull request merged**